lib/sycamore/sycamore/transforms/standardizer.py

```diff
@@ -0,0 +1,121 @@
+
+from sycamore.data import Document
+from sycamore.transforms.map import Map
+from dateutil import parser
+
+
+class Standardizer(ABC):
+
+    @abstractmethod
+    def standardize(self, document: Document) -> Document:
+        pass
+
+
+class LocationStandardizer(Standardizer):
+    """
+    This Class standardizes the format of US state location.
+    """
+
+    def __init__(self):
+        self._state_dict = {
+            "AK": "Alaska",
+            "AL": "Alabama",
+            "AR": "Arkansas",
+            "AZ": "Arizona",
+            "CA": "California",
+            "CO": "Colorado",
+            "CT": "Connecticut",
+            "DC": "District of Columbia",
+            "DE": "Delaware",
+            "FL": "Florida",
+            "GA": "Georgia",
+            "HI": "Hawaii",
+            "IA": "Iowa",
+            "ID": "Idaho",
+            "IL": "Illinois",
+            "IN": "Indiana",
+            "KS": "Kansas",
+            "KY": "Kentucky",
+            "LA": "Louisiana",
+            "MA": "Massachusetts",
+            "MD": "Maryland",
+            "ME": "Maine",
+            "MI": "Michigan",
+            "MN": "Minnesota",
+            "MO": "Missouri",
+            "MS": "Mississippi",
+            "MT": "Montana",
+            "NC": "North Carolina",
+            "ND": "North Dakota",
+            "NE": "Nebraska",
+            "NH": "New Hampshire",
+            "NJ": "New Jersey",
+            "NM": "New Mexico",
+            "NV": "Nevada",
+            "NY": "New York",
+            "OH": "Ohio",
+            "OK": "Oklahoma",
+            "OR": "Oregon",
+            "PA": "Pennsylvania",
+            "RI": "Rhode Island",
+            "SC": "South Carolina",
+            "SD": "South Dakota",
+            "TN": "Tennessee",
+            "TX": "Texas",
+            "UT": "Utah",
+            "VA": "Virginia",
+            "VT": "Vermont",
+            "WA": "Washington",
+            "WI": "Wisconsin",
+            "WV": "West Virginia",
+            "WY": "Wyoming",
+        }
+
+    def _standardize_state(self, state: str) -> str:
+        clean_state = state.lstrip().rstrip().upper()
+        return self._state_dict.get(clean_state, "")
+
+    def standardize(self, doc: Document) -> Document:
+        if "location" not in doc.properties["entity"]:
+            return doc
+        raw_location = doc.properties["entity"]["location"]
+        city, state = raw_location.split(",")
+        std_loc = f"{city}, {self._standardize_state(state)}"
+        doc.properties["entity"]["location"] = std_loc
+        return doc
+
+
+class DateTimeStandardizer(Standardizer):
+    """
+    This Class standardizes the format of dateTime.
+    Attributes:
+
+    """
+
+    def standardize(self, doc: Document) -> Document:
+        if "dateTime" not in doc.properties["entity"]:
+            return doc
+        raw_dateTime = doc.properties["entity"]["dateTime"]
+        raw_dateTime = raw_dateTime.replace("Local", "")
```

Comment:
This is also very brittle and only works in certain cases. That is probably OK for now as long as you document the limitations and ensure the code is robust to inputs not in the format you expect.